docs/python.md

```diff
@@ -47,22 +63,58 @@ If the output is a `pathlib.Path` object, that will be returned by the built-in
 To output `pathlib.Path` objects the file needs to exist, which means that you probably need to create a temporary file first. This file will automatically be deleted by Cog after it has been returned. For example:
 
 ```python
-def predict(self, input):
-    output = do_some_processing(input)
+def predict(self, image: Path = Input(description="Image to enlarge")) -> Path:
+    output = do_some_processing(image)
     out_path = Path(tempfile.mkdtemp()) / "my-file.txt"
     out_path.write_text(output)
     return out_path
 ```
 
-### `@cog.input(name, type, help, default=None, min=None, max=None, options=None)`
+## `Input(**kwargs)`
 
-The `@cog.input()` annotation describes a single input to the `predict()` function. The `name` must correspond to an argument name in `predict()`.
+Use cog's `Input()` function to define each of the parameters in your `predict()` method:
+
+```py
+class Predictor(BasePredictor):
+    def predict(self,
+            image: Path = Input(description="Image to enlarge"),
+            scale: float = Input(description="Factor to scale image by", default=1.5, gt=0, lt=10)
+    ) -> Path:
+```
 
-It takes these arguments:
+The `Input()` function takes these keyword arguments:
 
-- `type`: Either `str`, `int`, `float`, `bool`, or `Path` (be sure to add the import, as in the example above). `Path` is used for files. For more complex inputs, save it to a file and use `Path`.
-- `help`: A description of what to pass to this input for users of the model
+- `description`: A description of what to pass to this input for users of the model.
 - `default`: A default value to set the input to. If this argument is not passed, the input is required. If it is explicitly set to `None`, the input is optional.
-- `min`: A minimum value for `int` or `float` types.
-- `max`: A maximum value for `int` or `float` types.
-- `options`: A list of values to limit the input to. It can be used with `str`, `int`, and `float` inputs.
+- `gt`: For `int` or `float` types, the value must be greater than this number.
+- `ge`: For `int` or `float` types, the value must be greater than or equal to this number.
+- `lt`: For `int` or `float` types, the value must be less than this number.
+- `le`: For `int` or `float` types, the value must be less than or equal to this number.
+- `choices`: A list of possible values for this input.
+
+Each parameter of the `predict()` method must be annotated with a type. The supported types are:
+
+- `str`: a string
+- `int`: an integer
+- `float`: a floating point number
+- `bool`: a boolean
+- `cog.File`: a file-like object representing a file
+- `cog.Path`: a path to a file on disk
+
+## `Output(BaseModel)`
+
+Your `predict()` method can return a simple data type like a string or a number, or a more complex object with multiple values.
+
+You can optionally use cog's `Output()` function to define the object returned by your `predict()` method:
+
+```py
+from cog import BasePredictor, BaseModel
```

Comment:
This wasn't in there so I added it :D https://github.com/replicate/cog/pull/378/commits/7e989203e39ce2c21f5731187f7472894e5ffa1e

Reply:
Nice! I actually kinda knew that and hoped this document would will it into existence. :trollface: